MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  MovieQuizPresenter.swift
+//  MovieQuiz
+//
+//  Created by Victoria Isaeva on 10.05.2023.
+//
+
+
+import UIKit
+
+protocol MovieQuizViewControllerProtocol: AnyObject {
+    func show(quiz step: QuizStepViewModel)
+    func show(alertModel: AlertModel)
+    
+    func highlightImageBorder(isCorrectAnswer: Bool)
+    
+    func showLoadingIndicator()
+    func hideLoadingIndicator()
+    
+    func showNetworkError(message: String)
+}
+
+final class MovieQuizPresenter: QuestionFactoryDelegate {
+    
+    private let statisticService: StatisticService!
+    private var questionFactory: QuestionFactoryProtocol?
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    
+    private var currentQuestion: QuizQuestion?
+    private var correctAnswers: Int = 0
+    private  var currentQuestionIndex = 0
+    private let questionsAmount: Int = 10
+    
+    
+    init(viewController:MovieQuizViewControllerProtocol) {
```

Comment:
Очень важно замечать, где синтаксис языка начинает слипаться, и расставлять пробелы. Так ты скорее всего напишешь код в обще принятом аккуратном кодстайле 
```suggestion
    init(viewController: MovieQuizViewControllerProtocol) {
```